src/alpm_shelling.rs

```diff
@@ -0,0 +1,62 @@
+use crate::alpm_wrapper::AlpmWrapper;
+use itertools::Itertools;
+use std::process::{Command, Stdio};
+
+pub struct AlpmImpl {}
+
+pub fn new() -> AlpmImpl {
+	AlpmImpl {}
+}
+
+impl AlpmWrapper for AlpmImpl {
+	fn is_package_installed(&self, package: &str) -> bool {
```

Comment:
-T includes provides. So this probably does not do what you intend for it to.

Reply:
on the contrary, if something already provides a package in the system, I don't need to install it anew?

Reply:
That may be true, but the function name is still misleading. If this is intended behaviour then it should be renamed to `find_satisfier` or something.

Reply:
You're right about that. I've changed the name to `is_installed` now. This whole branch will probably be discarded eventually though.